lute/models/setting.py

```diff
@@ -225,3 +226,40 @@ def last_backup_display_date(self):
     def get_backup_settings():
         "Get BackupSettings."
         return BackupSettings()
+
+    @property
+    def time_since_last_backup(self):
+        """
+        Return the time since the last backup, or None if not set.
+        Eg. "3 days ago" or "40 seconds from now"
+        """
+        t = self.last_backup_datetime
+        if t is None:
+            return None
+
+        delta = int(time.time() - t)
+
+        thresholds = [
+            ("week", 1 * 60 * 60 * 24 * 7),
+            ("day", 1 * 60 * 60 * 24),
+            ("hour", 1 * 60 * 60),
+            ("minute", 1 * 60),
+            ("second", 1),
+        ]
+
+        for unit, seconds in thresholds:
+            multiples = abs(delta // seconds)
+            if multiples >= 1:
+                message = f"{multiples} {unit}"
+                if multiples > 1:
+                    message += "s"
+                break
+        else:
+            message = f"{abs(delta)} seconds"
+
+        if delta >= 0:
+            message += " ago"
+        else:
+            message += " from now"
```

Comment:
Why would time since last backup return a data in the future?  Weird date/time changes, daylight savings? :-)

If this execution path is valid, and I were a user, this would feel like a bug.  I'm not sure what the right thing to display would be if delta < 0.

Reply:
On WSL whenever my laptop goes to sleep it falls back in time and then I run a time-sync command. But if two commands run at the same time, then it throws my time into the future until I run a time-sync again.


So... its a contrived way to get into future time. One would have to run a backup in that duration and then correct time and then go see that it says "from now". DST and timezones wont affect it, however, since `time.time()` is Unix time I believe.


We could probably put a more safe value so users aren't confused by something saying it happened in the future or thinking that thats when the backup is scheduled for.


How about no human readable value if its in the future? It could still have the original timestamp there though.

Reply:
Sure, no human-readable thing, to prevent my computer from time travelling :-)

Reply:
Updated so that the human readable portion is hidden if the time is in the future.

![image](https://github.com/jzohrab/lute-v3/assets/9434751/86cca135-a107-4b14-8807-722a46edd52f)